src/qt/walletmodel.h

```diff
@@ -123,8 +123,8 @@ class WalletModel : public QObject
         // Copy constructor is disabled.
         UnlockContext(const UnlockContext&) = delete;
         // Move operator and constructor transfer the context
-        UnlockContext(UnlockContext&& obj) { CopyFrom(std::move(obj)); }
-        UnlockContext& operator=(UnlockContext&& rhs) { CopyFrom(std::move(rhs)); return *this; }
+        UnlockContext(UnlockContext&& obj) noexcept { CopyFrom(std::move(obj)); }
```

Comment:
I'm not sure this is correct. `CopyFrom` is not noexcept, so marking this as noexcept can be dangerous. 
CopyFrom seems to use the copy assignment operator with its `*this = rhs;` which too is not noexcept. I think it can currently be marked as noexcept (based on the members here), but this is dangerous; e.g. when you add any data that needs allocating, the `noexcept` would be a lie and bad_alloc exceptions would cause the program to abort

Reply:
@martinus 

The commented code is no longer relevant since https://github.com/bitcoin-core/gui/pull/711.

Nevertheless, I've dropped `performance-noexcept-move-constructor` commit for a follow up as it requires more thorough reviewing for similar cases.

Reply:
Good call, I find the `noexcept` stuff so brittle it's really hard to use. E.g. my changes to nanobench.h too were not completely correct because there's no guarantee that `std::string` move assignment is noexcept, see here: https://github.com/martinus/nanobench/pull/87